File: Nian_Karl/QL_Nian_Karl/src/org/uva/sea/ql/model/literal/IntegerLiteral.java
package org.uva.sea.ql.model.literal;

public class IntegerLiteral extends AbstractLiteral<Integer> {

	private final int intValue;
	
	public IntegerLiteral(int intValue) {
		this.intValue = intValue;
	}
	
	@Override
	public Integer getValue() {
		return intValue;
	}

}

File: CarlaErik/UvA.SoftCon.Questionnaire/AST/Literals/IntegerLiteral.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UvA.SoftCon.Questionnaire.AST.Expressions;

namespace UvA.SoftCon.Questionnaire.AST.Literals
{
    /// <summary>
    /// Represents a static, immutable integer value.
    /// </summary>
    public class IntegerLiteral : Literal<int>, IExpression
    {
        public IntegerLiteral(int value)
            : base(value) { }

        public override string ToString()
        {
            return Value.ToString();
        }
    }
}
